setup-env.sh

```diff
@@ -8,9 +8,19 @@ else
     machine="$( "$parent_dir"/src/machine.sh )"
 fi
 
-export SPACK_SYSTEM_CONFIG_PATH="$parent_dir"/sysconfigs/"$machine"
-export SPACK_USER_CONFIG_PATH="$parent_dir"/user-config
-export SPACK_USER_CACHE_PATH="$parent_dir"/user-cache
-. "$parent_dir"/spack/share/spack/setup-env.sh
+if [[ ${machine:0:1} == "/" ]]; then
+    uenv_mount=$machine
+    machine=uenv
+fi
```

Comment:
On what assumption do you check for `/`?

This breaks runs w/o uenvs. `. setup-env.sh` does not do anything on Balfrim atm.

Reply:
> On what assumption do you check for `/`?

A "normal" machine name cannot start with a `/`. If it's the case, we assume the user passed the path to the uenv mount point.

> This breaks runs w/o uenvs. . setup-env.sh does not do anything on Balfrim atm.

It's indeed broken but not because of that, rather the test on existing `pre-setup.sh`. I committed the fix (surround with explicit if statement)